structuredheadings/plugin.js

```diff
@@ -1,287 +1,294 @@
-/*eslint max-statements: [2, 50]*/ //I'll need to refactor somehow to meet this rule
-(function () {
-  CKEDITOR.plugins.add("structuredheadings", {
-    icons: "autonumberheading," +
-           "matchheading," +
-           "increaseheadinglevel," +
-           "decreaseheadinglevel," +
-           "restartNumbering",
-
-    init: function (editor) {
-      // list of elements allowed to be numbered
-      var allowedElements = {
-        h1: 1,
-        h2: 1,
-        h3: 1,
-        h4: 1,
-        h5: 1,
-        h6: 1
-      };
-
-      //some friendly setup for level changes
-      var headerList = ["h1", "h2", "h3", "h4", "h5", "h6"];
-      var firstHeaderKey = 0;
-      var lastHeaderKey = headerList.length - 1;
-
-      editor.addContentsCss(this.path + "styles/numbering.css");
-
-      // Helper function to set button state
-      var setCommandState = function (command, state) {
-        switch (state) {
-        case "on":
-          editor.getCommand(command).setState(CKEDITOR.TRISTATE_ON);
-          break;
-        case "off":
-          editor.getCommand(command).setState(CKEDITOR.TRISTATE_OFF);
-          break;
-        case "disabled":
-          editor.getCommand(command).setState(CKEDITOR.TRISTATE_DISABLED);
-          break;
-        }
-
-      };
-
-      var isNumbered = function (element) {
-        if (element.hasClass("autonumber")) {
-          return true;
+/*
+ * Helper Functions
+ */
+
+// list of elements allowed to be numbered
+var allowedElements = {
```

Comment:
Because this isn't a proper, I think you will need to wrap the entire file into an IIFE .  Otherwise I think these all just get shoved onto window